Src/Test.UnitTests.Sarif.PatternMatcher.Cli/AnalyzeCommandTests.cs

```diff
@@ -529,5 +742,50 @@ private static string GetSingleLineRuleDefinition()
 
             return JsonConvert.SerializeObject(definitions);
         }
+
+        private static string GetSingleLineRuleDefinitionFailureLevel(FailureLevel level)
+        {
+            string assemblyName = typeof(AnalyzeCommandTests).Assembly.Location;
+            assemblyName = Path.GetFileName(assemblyName);
+            var definitions = new SearchDefinitions()
+            {
+                ValidatorsAssemblyName = assemblyName,
+                SharedStringsFileName = "SharedStrings.txt",
+
+                Definitions = new List<SearchDefinition>(new[]
+                {
+                    new SearchDefinition()
+                    {
+                        Name = "FailureLevelTest", Id = "TEST001",
+                        Level = level,
+                        Message = "A problem occurred in '{0:scanTarget}'.",
+                        Description = "Failure Level Testing Rules",
+                        MatchExpressions = new List<MatchExpression>(new[]
+                        {
+                            new MatchExpression()
+                            {
+                                Id = "TEST001/001",
+                                Name ="NoValidation",
```

Comment:
>NoValidation [](http://example.com/codeflow?start=39&length=12)

Your test names do not encode sufficient information, please try to provide a more descriptive scenario.

'DoesNotWork',
'NoValidation'
'FailsQuickly'

these are general descriptions. 

'StaticOnly' static what? Pretend you are new to the team, what would you make of this?